feat: add blog post for security v1.9 (#262)
arcus-azure/arcus: community/blog/2022-12 Synchronous secret retrieval in Arcus Security v1.9/README.md

When using the secret providers, you can easily use them in asynchronous contexts like an API controller or upon each service method call. The problem happens, though, when you want to register a service in the dependency container. If that service needs to be initialized upon creation, you need the secret at creation time. Dependency containers usually do not work with asynchronous registration ([the one from Microsoft does not](https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines#recommendations)).

The following workaround usually happens when the Arcus secret store is used in such scenarios:
```csharp
using Microsoft.Extensions.DependencyInjection;
using Arcus.Security.Core;

public void ConfigureServices(IServiceCollection services)
{
    services.AddSingleton(serviceProvider =>
    {
        var secretProvider = serviceProvider.GetRequiredService<ISecretProvider>();
        string secretValue = secretProvider.GetRawSecretAsync("MySecret").GetAwaiter().GetResult();

        return new MyServiceThatRequiresSecretRightAway(secretValue);
    });
}
```

## The solution of sync-support secret providers

> What we did in Arcus Security v1.9, is something that is both simple and very powerful - and without any breaking changes. We introduced a new `ISyncSecretProvider` interface that inherits from our main `ISecretProvider` interface. This means that every synchronous secret provider is also an asynchronous secret provider. The transition from sync to async is a lot more straightforward than the way around. After that, we checked our available built-in secret providers and used the synchronous version on each one that could support synchronous secret retrieval. This includes Azure Key Vault, environment variables, configuration, User Secrets, Docker Secrets and others.

Reviewer: ```suggestion
What we did in Arcus Security v1.9, is something that is both simple and very powerful - and without any breaking changes. We introduced a new `ISyncSecretProvider` interface that inherits from our main `ISecretProvider` interface. This means that every synchronous secret provider is also an asynchronous secret provider. The transition from sync to async is a lot more straightforward than the other way around. After that, we checked our available built-in secret providers and used the synchronous version on each one that could support synchronous secret retrieval. This includes Azure Key Vault, environment variables, configuration, User Secrets, Docker Secrets and others.
```

> What we did in Arcus Security v1.9, is something that is both simple and very powerful - and without any breaking changes. We introduced a new `ISyncSecretProvider` interface that inherits from our main `ISecretProvider` interface. This means that every synchronous secret provider is also an asynchronous secret provider. The transition from sync to async is a lot more straightforward than the other way around. After that, we checked our available built-in secret providers and used the synchronous version on each one that could support synchronous secret retrieval. This includes Azure Key Vault, environment variables, configuration, User Secrets, Docker Secrets and others.

Reviewer 2: You're saying here that 'every sync secret provider is also an async secret provider, but before saying that, I think it would be good if we could first say what this ISync... interface is about, without going into too much detail.

> What we did in Arcus Security v1.9, is something that is both simple and very powerful - and without any breaking changes. We introduced a new `ISyncSecretProvider` interface that inherits from our main `ISecretProvider` interface. This means that every synchronous secret provider is also an asynchronous secret provider. The transition from sync to async is a lot more straightforward than the other way around. After that, we checked our available built-in secret providers and used the synchronous version on each one that could support synchronous secret retrieval. This includes Azure Key Vault, environment variables, configuration, User Secrets, Docker Secrets and others.

Reviewer 2: ```suggestion
What we did in Arcus Security v1.9, is something that is both simple and very powerful - and without any breaking changes. We introduced a new `ISyncSecretProvider` interface that defines the contract to retrieve secrets in a synchronous fashion.  The `ISyncSecretProvider` inherits from our main `ISecretProvider` interface which means that every synchronous secret provider is also an asynchronous secret provider. The transition from sync to async is a lot more straightforward than the other way around. After that, we checked our available built-in secret providers and used the synchronous version on each one that could support synchronous secret retrieval. This includes Azure Key Vault, environment variables, configuration, User Secrets, Docker Secrets and others.
```

Author: Yes, this is better. Thx!

Our sample now looks like this:
```csharp
using Microsoft.Extensions.DependencyInjection;
using Arcus.Security.Core;

public void ConfigureServices(IServiceCollection services)
{
    services.AddSingleton(serviceProvider =>
    {
        var secretProvider = serviceProvider.GetRequiredService<ISecretProvider>();
        string secretValue = secretProvider.GetRawSecret("MySecret");

        return new MyServiceThatRequiresSecretRightAway(secretValue);
    });
}
```

Note that we still use the `ISecretProvider` and do not need to add any other type of method.
✨ Like magic, all your code now supports synchronous secret retrieval.

## Implementing your own synchronous secret provider
The Arcus secret store was from its inception very flexible and extensible and that is also why we made a new interface in the same way we have the `ICachedSecretProvider` and `IVersionedSecretProvider`. This lets consumers choose if they want their secret provider to support synchronous secret retrieval or not. If not, then the secret store will either check for a different registration that does support it or throw a `NotSupportedException` if none does.

> Implementing a secret provider that does check the Windows Registry for secrets could look like this:

Reviewer: ```suggestion
Implementing a secret provider that checks the Windows Registry for secrets could look like this:
```
